src/wifipaf/WiFiPAFLayer.h

```diff
@@ -0,0 +1,204 @@
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "WiFiPAFConfig.h"
+#include "WiFiPAFEndPoint.h"
+#include "WiFiPAFLayerDelegate.h"
+#include "WiFiPAFRole.h"
+#include <lib/core/CHIPError.h>
+#include <lib/support/DLLUtil.h>
+#include <system/SystemLayer.h>
+#include <system/SystemPacketBuffer.h>
+
+namespace chip {
+namespace WiFiPAF {
+
+/**
+ *  @def NUM_PAFTP_SUPPORTED_PROTOCOL_VERSIONS
+ *
+ *  Number of unsigned 4-bit representations of supported transport protocol
+ *  versions encapsulated in a BleTransportCapabilitiesRequest. Defined by CHIP
+ *  over PAFTP protocol specification.
+ */
+#define NUM_PAFTP_SUPPORTED_PROTOCOL_VERSIONS 8
+
+// Version(s) of the CHIP PAF Transport Protocol that this stack supports.
+#define CHIP_PAF_TRANSPORT_PROTOCOL_MIN_SUPPORTED_VERSION kWiFiPAFTransportProtocolVersion_V1
+#define CHIP_PAF_TRANSPORT_PROTOCOL_MAX_SUPPORTED_VERSION kWiFiPAFTransportProtocolVersion_V1
+
+/// Enum defining versions of CHIP over PAF transport protocol.
+typedef enum
+{
+    kWiFiPAFTransportProtocolVersion_None = 0,
+    kWiFiPAFTransportProtocolVersion_V1   = 4 // PAFTP as defined by CHIP v1.5
+} WiFiPAFTransportProtocolVersion;
+
+inline constexpr size_t kCapabilitiesRequestMagicnumLength          = 2;
+inline constexpr size_t kCapabilitiesRequestL2capMtuLength          = 2;
+inline constexpr size_t kCapabilitiesRequestSupportedVersionsLength = 4;
+inline constexpr size_t kCapabilitiesRequestWindowSizeLength        = 1;
+constexpr size_t kCapabilitiesRequestLength = (kCapabilitiesRequestMagicnumLength + kCapabilitiesRequestL2capMtuLength +
+                                               kCapabilitiesRequestSupportedVersionsLength + kCapabilitiesRequestWindowSizeLength);
+
+inline constexpr size_t kCapabilitiesResponseMagicnumLength                = 2;
+inline constexpr size_t kCapabilitiesResponseL2capMtuLength                = 2;
+inline constexpr size_t kCapabilitiesResponseSelectedProtocolVersionLength = 1;
+inline constexpr size_t kCapabilitiesResponseWindowSizeLength              = 1;
+constexpr size_t kCapabilitiesResponseLength(kCapabilitiesResponseMagicnumLength + kCapabilitiesResponseL2capMtuLength +
+                                             kCapabilitiesResponseSelectedProtocolVersionLength +
+                                             kCapabilitiesResponseWindowSizeLength);
+
+class PAFTransportCapabilitiesRequestMessage
+{
+public:
+    /**
+     * An array of size NUM_PAFTP_SUPPORTED_PROTOCOL_VERSIONS listing versions of the
+     * PAF transport protocol that this node supports. Each protocol version is
+     * specified as a 4-bit unsigned integer. A zero-value represents unused
+     * array elements. Counting up from the zero-index, the first zero-value
+     * specifies the end of the list of supported protocol versions.
+     */
+    uint8_t mSupportedProtocolVersions[(NUM_PAFTP_SUPPORTED_PROTOCOL_VERSIONS / 2) + (NUM_PAFTP_SUPPORTED_PROTOCOL_VERSIONS % 2)];
```

Comment:
See, here we compute from NUM_PAFTP_SUPPORTED_PROTOCOL_VERSIONS.

The real invariant is that `sizeof(mSupportedProtocolVersions) == kCapabilitiesRequestSupportedVersionsLength`, so why are we not just setting one from the other?  i.e. moving all the constants to be declared in this message class, etc.